frontend/appflowy_flutter/lib/workspace/presentation/widgets/pop_up_action.dart

```diff
@@ -156,24 +158,30 @@ class ActionCellWidget<T extends PopoverAction> extends StatelessWidget {
   }
 }
 
-class PopoverActionCellWidget<T extends PopoverAction> extends StatelessWidget {
-  PopoverActionCellWidget({
+class PopoverActionCellWidget<T extends PopoverAction> extends StatefulWidget {
+  const PopoverActionCellWidget({
     Key? key,
-    this.mutex,
-    // required this.popoverController,
+    required this.popoverController,
     required this.action,
     required this.itemHeight,
   }) : super(key: key);
 
   final T action;
   final double itemHeight;
 
-  final PopoverMutex? mutex;
-  final PopoverController popoverController = PopoverController();
+  final PopoverController popoverController;
+
+  @override
+  State<PopoverActionCellWidget> createState() =>
+      _PopoverActionCellWidgetState();
+}
 
+class _PopoverActionCellWidgetState<T extends PopoverAction>
+    extends State<PopoverActionCellWidget<T>> {
+  final popoverController = PopoverController();
   @override
   Widget build(BuildContext context) {
-    final actionCell = action as PopoverActionCell;
+    final actionCell = widget.action as PopoverActionCell;
     final leftIcon =
         actionCell.leftIcon(Theme.of(context).colorScheme.onSurface);
     final rightIcon =
```

Comment:
Should we move these to initState to avoid rebuilds to cause bugs? I don't know enough about the popover to say that it can happen, but it's a common trap for stateful widgets.

Reply:
It's stateless widget before. I changed it to stateful to solve a popover state bug.

Reply:
Can you point out what bugs may occur?